testrpc/testrpc.py

```diff
@@ -83,13 +154,94 @@ def format_block_number(block_number):
 
     return block_number
 
+class LogListener(object):
+    def __init__(self):
+        self.log_idx = 0
+
+    def __call__(self, log):
+        block = evm.block
+        if block.number not in event_log:
+            event_log[block.number] = []
+
+        event_log[block.number].append({
+            "log": log, "log_idx": self.log_idx,
+            "tx_idx": len(block.get_transactions()), "txhash": evm.last_tx.hash,
+            "pending": False, "block": block
+        })
+        self.log_idx += 1
+
+def mine(evm):
+    if evm.block.number == 0:
+        evm.block.log_listeners.append(LogListener())
+
+    listener = evm.block.log_listeners.pop(0)
```

Comment:
No, but I can see why you'd be concerned. The "mine" function on the block doesn't actually log anything. All it does is instantiate a new "block" object to collect transactions in. The events are logged when the transactions come in, not when they're mined. That's not how it works on the real blockchain, of course, so I'll go ahead and change this bit to be less confusing. I ultimately don't really have a good reason for popping the listeners off the block, apart from trying to save some memory as the blockchain grows, which is totally a premature optimization. And it looks like PyEthereum carries forward the log listeners upon mining, so I should be able to remove the `append` line as well.